### ISSUE TITLE
Loading rules from a JSON file does not work when following the documentation

### ISSUE THREAD

**Issue by nidkil** [stalniy/casl#177]
I was changing the setup of the ability rules from DSL to loading them from a JSON file. To setup the initialization from JSON I was following the section `Store ability templates` in the documentation [here](https://stalniy.github.io/casl/abilities/storage/2017/07/22/storing-abilities.html).

Loading the rule templates from the JSON file, initializing the template variables and passing them to Ability worked fine. But once I started checking the rules they did not work as expected. I compared the DSL version to the JSON version using console.log(ability.rules) and could not find any difference.

It turns out the there was a difference between the way the DSL and JSON version were being initialized.

In the DSL version I was following a Medium post by Sergii, which initialized the rules in the following way:

```
return new Ability(rules, { subjectName });
```

This is the subjectName function:

```
const TYPE_KEY = Symbol.for('type');

function subjectName(subject) {
  if (!subject || typeof subject === 'string') {
    return subject;
  }
  return subject[TYPE_KEY];
}
```

In the JSON template version I was following the section `Store ability templates` from the documentation [here](https://stalniy.github.io/casl/abilities/storage/2017/07/22/storing-abilities.html), which was initializing the rules without the subjectName:

```
return new Ability(rules);
```

After adding `subjectName` to the initialization the rules are working as expected. I do not really understand what `subjectName` is doing. Could you please explain?

Should the documentation be updated?

**Comment by stalniy** [comment]
Some example would be helpful

**Comment by nidkil** [comment]
Of the code? All the code is doing is converting JSON rules to Javascript objects and passing them to Ability. There is no difference between the rules created with DSL and JSON. That was the frustrating part.

**Comment by tmikaeld** [comment]
I think he means this:

![Image 2019-04-12 at 10 30 58 AM](https://user-images.githubusercontent.com/907195/56023324-e6398c80-5d0d-11e9-8cb1-0b069e8b5a3e.png)

**Comment by nidkil** [comment]
Ah sorry. In my head I had already shared that :-)

**Comment by tmikaeld** [comment]
@nidkil You can edit the description of the issue and describe what was the difference, because it's hard to track issues that have no proper description.

**Comment by nidkil** [comment]
What happened to the discription!?!?!?!?

**Comment by nidkil** [comment]
For some strange reason the discription was not saved. Sorry. Should be visible now.

**Comment by tmikaeld** [comment]
It is

**Comment by stalniy** [comment]
`subjectName` is briefly described in [Instance Checks](https://stalniy.github.io/casl/abilities/2017/07/21/check-abilities.html#instance-checks). Let me know if that description is not enough or how you would rephrase it

**Comment by nidkil** [comment]
Hi Stalniy, I did not see this description before. It is clear. No need to rephrase it.

**Comment by stalniy** [comment]
Ok, closing this
